In [1]:
import numpy as np
import pandas as pd
import pandas_profiling
import matplotlib.pyplot as plt

In [3]:
Spotify = pd.read_csv('SpotifyFeatures.csv')
lyrics = pd.read_csv('only_lyrics.csv')

FileNotFoundError: [Errno 2] File b'SpotifyFeatures.csv' does not exist: b'SpotifyFeatures.csv'

In [3]:
Spotify.head(1)

genre     artist_name                   track_name                track_id  \
0  Movie  Henri Salvador  C'est beau de faire un Show  0BRjO6ga9RKCKjfDqeFgWV   

   popularity  acousticness  danceability  duration_ms  energy  \
0           0         0.611         0.389        99373    0.91   

   instrumentalness key  liveness  loudness   mode  speechiness    tempo  \
0               0.0  C#     0.346    -1.828  Major       0.0525  166.969   

  time_signature  valence  
0            4/4    0.814

In [4]:
lyrics.head(1)

artist                   song                                        link  \
0   ABBA  Ahe's My Kind Of Girl  /a/abba/ahes+my+kind+of+girl_20598417.html   

                                                text  
0  Look at her face, it's a wonderful face  \nAnd...

## Merging lyrics and the songs

In [5]:
Spotify['track_name']=Spotify['track_name'].str.lower()
Spotify['track_name']=Spotify['track_name'].str.strip()
Spotify['artist_name']=Spotify['artist_name'].str.lower()
Spotify['artist_name']=Spotify['artist_name'].str.strip()
lyrics['song']=lyrics['song'].str.lower()
lyrics['song']=lyrics['song'].str.strip()
lyrics['artist']=lyrics['artist'].str.lower()
lyrics['artist']=lyrics['artist'].str.strip()

In [9]:
song_lyrics = Spotify.merge(lyrics, left_on=['track_name', 'artist_name'], right_on=['song', 'artist'])  

In [16]:
song_lyrics =song_lyrics.sort_values(by=['popularity'],ascending=False)
song_lyrics.head(1)

genre artist_name  track_name                track_id  popularity  \
6583   Pop  ed sheeran  photograph  1HNkqx9Ahdgi1Ixy2xkKkL          86   

      acousticness  danceability  duration_ms  energy  instrumentalness  ...  \
6583         0.607         0.614       258987   0.379          0.000464  ...   

     loudness   mode  speechiness    tempo  time_signature  valence  \
6583   -10.48  Major       0.0476  107.989             4/4    0.201   

          artist        song                                    link  \
6583  ed sheeran  photograph  /e/ed+sheeran/photograph_21058341.html   

                                                   text  
6583  [Verse 1]  \nLoving can hurt, loving can hurt ...  

[1 rows x 22 columns]

In [18]:
song_lyrics=song_lyrics.drop(['artist', 'song', 'link'], axis=1)
song_lyrics.shape

KeyError: "['artist' 'song' 'link'] not found in axis"

In [19]:
song_lyrics.shape

(8472, 19)

#### There are 8472 songs in the dataframe when I merge the databases with song features and the lyrics in it.
#### But, there are multiple repetitive songs with different genre

In [ ]:
Profile = pandas_profiling.ProfileReport(Sportify)
Profile.to_file('Sportify_report.html')

In [ ]:
Sportify2 = pd.read_csv('SONG WITH LYRICS.csv', index_col=0)
Sportify2['track_name'] = Sportify2['track_name'].str.lower()
Sportify2['artist_name'] = Sportify2['artist_name'].str.lower()

In [ ]:
Track_year = pd.read_csv('billboard_lyrics_1964-2015.csv', sep=';',index_col=0)
Track_year.head()

In [ ]:
Updatewithyear_Lyrics = Sportify2.merge(Track_year, how='left', left_on=['track_name', 'artist_name'], right_on=['Song', 'Artist'])


In [ ]:
Updatewithyear_Lyrics

In [ ]:
Updatewithyear_Lyrics['Year'].value_counts()